python/src/equistore/operations/split.py

```diff
@@ -0,0 +1,211 @@
+    with the dimensions of the blocks reduced to only contain the specified
+    indices for the corresponding group.
+
+    For example, to split a block along the "samples" axis, according to the
+    "structure" index, where structures 0, 6, and 7 are in the first returned
+    :py:class`TensorMap`; 2, 3, and 4 in the second; and 1, 5, 8, 9, and 10 in
+    the third:
+
+    .. code-block:: python
+
+        split_tensorblock = split_block(
+            block,
+            axis="samples",
+            grouped_idxs=[
+                Labels(
+                    names=["structure"],
+                    values=np.array([[0], [6], [7]])
+                ),
+                Labels(
+                    names=["structure"],
+                    values=np.array([[2], [3], [4]])
+                ),
+                Labels(
+                    names=["structure"],
+                    values=np.array([[1], [5], [8], [9], [10]])
+                ),
+            ]
+        )
+
+    :param block: a :py:class:`TensorBlock` to be split
+    :param axis: a str, either "samples" or "properties", that indicates the
+        :py:class:`TensorBlock` axis along which the named index (or indices) in
+        ``grouped_idxs`` belongs. Each :py:class:`TensorBlock` returned could
+        have a reduced dimension along this axis, but the other axes will remain
+        the same size.
+    :param grouped_idxs: a list of :py:class:`Labels` containing the names and
+        values of the indices along the specified ``axis`` which should be in
+        each respective output :py:class:`TensorBlock`.
+
+    :return: a list of:py:class:`TensorBlock` that corresponds to the split
+        input ``block``. Each block in the returned list contains only the named
+        indices in the respective py:class:`Labels` object of ``grouped_idxs``.
+    """
+    # Check input args
+    if not isinstance(block, TensorBlock):
+        raise TypeError("``block`` should be an equistore ``TensorBlock``")
+    _check_args(block, axis, grouped_idxs)
+
+    return _split_block(block, axis, grouped_idxs)
+
+
+def _split_block(
+    block: TensorBlock,
+    axis: str,
+    grouped_idxs: List[Labels],
+) -> List[TensorBlock]:
+    """
+    Splits a TensorBlock into mutliple blocks, as in the public function
+    :py:func:`split_block` but with no input checks. Note that the block is
+    currently split into N new blocks by performing N number of slice
+    operations. There may be a more efficient way of doing it, but this is not
+    yet implemented.
+    """
+    new_blocks = []
+    for indices in grouped_idxs:
+        # Perfom the slice, either along the samples or properties axis
+        if axis == "samples":
+            new_block = _slice_block(block, samples=indices)
+        else:  # properties
+            new_block = _slice_block(block, properties=indices)
+        new_blocks.append(new_block)
+
+    return new_blocks
+
+
+def _check_args(
```

Comment:
Why is there another check_args function here? The name might be confusing with the one from the slice functions even of they are in different files.

Reply:
They are called the same thing but do different things in ``slice.py`` and ``split.py``. I can change the names but don't see much point as they just check arguments in each respective file and should be hidden